db/rdbms/src/main/java/io/camunda/db/rdbms/write/service/AuditLogWriter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.db.rdbms.write.service;
+
+import io.camunda.db.rdbms.config.VendorDatabaseProperties;
+import io.camunda.db.rdbms.sql.AuditLogMapper;
+import io.camunda.db.rdbms.write.domain.AuditLogDbModel;
+import io.camunda.db.rdbms.write.queue.ContextType;
+import io.camunda.db.rdbms.write.queue.ExecutionQueue;
+import io.camunda.db.rdbms.write.queue.QueueItem;
+import io.camunda.db.rdbms.write.queue.WriteStatementType;
+
+public class AuditLogWriter {
+
+  private final ExecutionQueue executionQueue;
+  private final AuditLogMapper mapper;
+  private final VendorDatabaseProperties vendorDatabaseProperties;
+
+  public AuditLogWriter(
+      final ExecutionQueue executionQueue,
+      final AuditLogMapper mapper,
+      final VendorDatabaseProperties vendorDatabaseProperties) {
+    this.executionQueue = executionQueue;
+    this.mapper = mapper;
+    this.vendorDatabaseProperties = vendorDatabaseProperties;
+  }
+
+  public void create(final AuditLogDbModel auditLog) {
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.AUDIT_LOG,
+            WriteStatementType.INSERT,
+            auditLog.entityKey(),
+            "io.camunda.db.rdbms.sql.AuditLogMapper.insert",
+            truncateDetails(auditLog)));
+  }
+
+  private AuditLogDbModel truncateDetails(final AuditLogDbModel auditLog) {
+    // Truncate details field if it exceeds database limits
+    final String details = auditLog.details();
+    if (details != null && details.length() > vendorDatabaseProperties.errorMessageSize()) {
+      final String truncated =
+          details.substring(0, vendorDatabaseProperties.errorMessageSize()) + "... (truncated)";
```

Comment:
🤔 The idea is to put serialized data of varying length here. If we do this, it will break it. It might be better that we define it as a "blob" if possible.